src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -63,12 +73,13 @@ public class IotDataPlaneClientFactory {
      */
     @Inject
     public IotDataPlaneClientFactory(DeviceConfiguration deviceConfiguration) {
-        configureClient(deviceConfiguration);
+        this.deviceConfiguration = deviceConfiguration;
+        configureClient(true);
         deviceConfiguration.onAnyChange((what, node) -> {
             if (validString(node, DEVICE_PARAM_AWS_REGION) || validPath(node, DEVICE_PARAM_ROOT_CA_PATH) || validPath(
                     node, DEVICE_PARAM_CERTIFICATE_FILE_PATH) || validPath(node, DEVICE_PARAM_PRIVATE_KEY_PATH)
                     || validString(node, DEVICE_PARAM_IOT_DATA_ENDPOINT)) {
-                configureClient(deviceConfiguration);
+                configureClient(true);
```

Comment:
this will block the main context thread due to retries. We must not block

Reply:
Totally removed client configuration in constructor and subscribe as suggested. Setting the client to null in any of these cases. May have to make it Atomic and optional.

Reply:
Atomic+Optional is totally unnecessary